SoftLayer/CLI/block/detail.py

```diff
@@ -5,16 +5,35 @@
 import SoftLayer
 from SoftLayer.CLI import environment
 from SoftLayer.CLI import formatting
+from SoftLayer.CLI import helpers
 from SoftLayer import utils
 
 
+def get_block_volume_id(volume_id, block_manager):
+    """Returns the volume id.
+
+    :param volume_id: ID of volume.
+    :param block_manager: Block Storage Manager.
+    :return: Returns the volume id.
+    """
+    storage_list = block_manager.list_block_volumes()
+    for storage in storage_list:
+        if volume_id == storage['username']:
+            volume_id = storage['id']
+            break
+
+    return volume_id
+
+
 @click.command()
 @click.argument('volume_id')
 @environment.pass_env
 def cli(env, volume_id):
     """Display details for a specified volume."""
     block_manager = SoftLayer.BlockStorageManager(env.client)
-    block_volume = block_manager.get_block_volume_details(volume_id)
+    volume_id = get_block_volume_id(volume_id, block_manager)
```

Comment:
since were moving get_block_volume_id to the block manager's resolvers feature, we can just remove this line and the rest should work.